src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets

```diff
@@ -56,6 +56,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <GarbageCollectionAdaptationMode Condition="'$(GarbageCollectionAdaptationMode)' == '' And '$(ServerGarbageCollection)' == 'true'">1</GarbageCollectionAdaptationMode>
   </PropertyGroup>
 
+  <!-- Default settings for trimmed apps. -->
+  <PropertyGroup Condition="'$(PublishTrimmed)' == 'true'">
```

Comment:
1. I think we should turn both SGs on by default for both Trimmed and Aot. cc @layomia @ericstj 
2. I think it would make sense to put a "trimmed or aot" section in

```xml
  <!-- Default settings for trimmed and aot'd apps. -->
  <PropertyGroup Condition="'$(PublishTrimmed)' == 'true' Or '$(PublishAot)' == 'true'">
    <!-- Enable the built-in source generators. -->
    <EnableRequestDelegateGenerator Condition="'$(EnableRequestDelegateGenerator)' == ''">true</EnableRequestDelegateGenerator>
    <EnableConfigurationBindingGenerator Condition="'$(EnableConfigurationBindingGenerator)' == ''">true</EnableConfigurationBindingGenerator>
  </PropertyGroup>

  <!-- Default settings for aot'd apps. -->
  <PropertyGroup Condition="'$(PublishAot)' == 'true'">
    <!-- Enable System.GC.DynamicAdaptationMode by default for aot'd apps to get a GC that dynamically adapts to the application's memory use. -->
    <GarbageCollectionAdaptationMode Condition="'$(GarbageCollectionAdaptationMode)' == '' And '$(ServerGarbageCollection)' == 'true'">1</GarbageCollectionAdaptationMode>
  </PropertyGroup>
```